tests/test_fock.py

```diff
@@ -160,143 +127,375 @@ def circuit(x):
-                    sf.ops.Xgate(0.2) | q[0]
-                    sf.ops.S2gate(0.1) | q
-
-                state = eng.run(prog).state
-                return sfop(state, wires, args)[0]
-
-            if op.num_params == 0:
-                self.assertAllEqual(circuit(), SF_reference())
-            elif op.num_params == 1:
-                if g == 'FockStateProjector':
-                    p = np.array([1])
-                else:
-                    p = a_array if op.par_domain == 'A' else a
-                self.assertAllEqual(circuit(p), SF_reference(p))
-
-    def test_polyxp(self):
-        """Test that PolyXP works as expected"""
-        self.logTestName()
+        b = 0.123
+        c = 0.532
+        wires = [0]
+
+        gate_name = "CatState"
+        operation = qml.CatState
+
+        cutoff_dim = 10
+        dev = qml.device("strawberryfields.fock", wires=2, cutoff_dim=cutoff_dim)
+
+        sf_operation = dev._operation_map[gate_name]
 
+        assert dev.supports_operation(gate_name)
+
+        @qml.qnode(dev)
+        def circuit(*args):
+            qml.TwoModeSqueezing(0.1, 0, wires=[0, 1])
+            operation(*args, wires=wires)
+            return qml.expval(qml.NumberOperator(0)), qml.expval(qml.NumberOperator(1))
+
+        res = circuit(a, b, c)
+        sf_res = SF_gate_reference(sf_operation, cutoff_dim, wires, a * np.exp(1j * b), c)
+        assert np.allclose(res, sf_res, atol=tol, rtol=0)
+
+    def test_tensorn_not_supported(self):
+        """Test error is raised with the unsupported TensorN observable"""
+        dev = qml.device("strawberryfields.fock", wires=2, cutoff_dim=2)
+
+        observable = qml.TensorN
+        wires = [0, 1]
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.expval(observable(wires=wires))
+
+        with pytest.raises(
+            qml.DeviceError,
+            match="Observable TensorN not supported " "on device strawberryfields.fock",
+        ):
+            circuit()
+
+
+class TestExpectation:
+    """Test that all supported expectations work as expected when compared to
+    the Strawberry Fields results"""
+
+    def test_number_operator(self, tol):
+        """Test that the expectation value of the NumberOperator observable
+        yields the correct result"""
+        cutoff_dim = 10
+
+        dev = qml.device("strawberryfields.fock", wires=2, cutoff_dim=cutoff_dim)
+
+        gate_name = "NumberOperator"
+        assert dev.supports_observable(gate_name)
+
+        op = qml.NumberOperator
+        sf_expectation = dev._observable_map[gate_name]
+        wires = [0]
+
+        @qml.qnode(dev)
+        def circuit(*args):
+            qml.Displacement(0.1, 0, wires=0)
+            qml.TwoModeSqueezing(0.1, 0, wires=[0, 1])
+            return qml.expval(op(*args, wires=wires))
+
+        assert np.allclose(
+            circuit(), SF_expectation_reference(sf_expectation, cutoff_dim, wires), atol=tol, rtol=0
```

Comment:
Here, is it implicit that `SF_expectation_reference` uses the same gates as the QNode above?

Reply:
Good question! I've now included a transition to using the [`Dgate`](https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.Dgate.html) instead of `Xgate` from SF

Originally I was mostly working with previous tests
https://github.com/XanaduAI/pennylane-sf/blob/c47d733fa77e3b0c6241f451da65743af98e094d/tests/test_fock.py#L259